app/controllers/api/v1/discussions_users_controller.rb

```diff
@@ -0,0 +1,43 @@
+class API::V1::DiscussionsUsersController < API::V1::ApplicationController
+
+  before_action :load_discussion, only: [:create, :destroy]
+  before_action :load_user, only: [:create, :destroy]
+  before_action :check_prior_rsvp, only: [:create]
+  before_action :check_rsvp_exists, only: [:destroy]
+
+  def create
+    if @user.discussions_users.create(discussion_id: @discussion.id)
```

Comment:
A chance of error is there when we have used different foreign key names, or any other customization.